python/tests/expressions/test_visitors.py

```diff
@@ -1429,3 +1432,162 @@ def test_rewrite_bound():
+
+@pytest.fixture
+def spec() -> PartitionSpec:
+    return PartitionSpec(PartitionField(1, 1000, IdentityTransform(), "id"))
+
+
+def test_identity_projection(schema: Schema, spec: PartitionSpec):
+    predicates = [
+        NotNull("id"),
+        IsNull("id"),
+        LessThan("id", 100),
+        LessThanOrEqual("id", 101),
+        GreaterThan("id", 102),
+        GreaterThanOrEqual("id", 103),
+        EqualTo("id", 104),
+        NotEqualTo("id", 105),
+    ]
+
+    expected = [
+        BoundNotNull(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+        BoundIsNull(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+        BoundLessThan(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(100),
+        ),
+        BoundLessThanOrEqual(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(101),
+        ),
+        BoundGreaterThan(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(102),
+        ),
+        BoundGreaterThanOrEqual(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(103),
+        ),
+        BoundEqualTo(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(104),
+        ),
+        BoundNotEqualTo(
+            term=BoundReference[int](
+                field=NestedField(field_id=1000, name="id", field_type=IntegerType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=literal(105),
+        ),
+    ]
+
+    for idx in range(len(predicates)):
+        expr = inclusive_projection(predicates[idx], schema, spec)
+        assert expr == expected[idx]
+
+
+def test_identity_projection_ignore_casing(schema: Schema, spec: PartitionSpec):
```

Comment:
This could use a few more test cases:

* One with each transform
* At least one `And` where one child can be projected and the other cannot
* At least one `Or` where one child can be projected and the other cannot
* At least one with `Not` that gets rewritten